### PR TITLE
2024 documentation update on GPU wording

### PR DESCRIPTION
**Pull request by napetrov**
# Description
Updating documentation for 2024.0 release, related to https://github.com/oneapi-src/oneDAL/pull/2480

Changes proposed in this pull request:
- Cleaning GPU support and DAAL cpp sycl from docs as deprecated and removed
- Adding oneDAL intro wording and supported algorithms
- Adding oneDAL GPU support section

### CODE REVIEW

File: docs/source/onedal/introduction.rst [hunk 1]
```diff
@@ -16,10 +16,51 @@
 
 Introduction
 ************
+oneDAL provides redesizgned version of interfaces that accounts for multi-device targets
```
Comment (ethanglaser) [inline docs/source/onedal/introduction.rst:19]:
```suggestion
oneDAL provides redesigned versions of interfaces that account for multi-device targets
```

File: docs/source/onedal/introduction.rst [hunk 2]
```diff
@@ -16,10 +16,51 @@
...
+such as CPU and GPU, distributed SPMD interfaces and many more.
+
+Algorithms support
+*****************
+
+For the following algorithms are avaialbe in oneDAL:
```
Comment (ethanglaser) [inline docs/source/onedal/introduction.rst:25]:
```suggestion
The following algorithms are available in oneDAL:
```

File: docs/source/daal/cpu-vs-gpu.rst [hunk 3]
```diff
@@ -19,90 +19,5 @@
...
-       - On GPU, the ``weights`` parameter is not supported.
-   * - :ref:`kernel`
-     - On GPU, the only supported computation mode (``ComputationMode``) is ``matrixMatrix``.
-   * - :ref:`objective_function`
-     -
-       - On GPU, only :ref:`logistic_loss` and :ref:`cross_entropy_loss` are supported, :ref:`mse` is not supported.
-       - On GPU, ``resultsToCompute`` only computes ``value``, ``gradient``, and ``hessian``.
-   * - :ref:`logistic_regression`
-     - ``penaltyL1`` is not supported on GPU
+GPU support in DAAL part of library have been deprecated and removed from library.
```
Comment (ethanglaser) [inline docs/source/daal/cpu-vs-gpu.rst:22]:
```suggestion
GPU support in DAAL part of library has been deprecated and removed from library.
```

**Review by ethanglaser — COMMENTED** [review]
Will take another look when onedal gpu support section added

File: docs/source/daal/cpu-vs-gpu.rst [hunk 4]
```diff
@@ -19,90 +19,5 @@
...
-       - On GPU, the ``weights`` parameter is not supported.
-   * - :ref:`kernel`
-     - On GPU, the only supported computation mode (``ComputationMode``) is ``matrixMatrix``.
-   * - :ref:`objective_function`
-     -
-       - On GPU, only :ref:`logistic_loss` and :ref:`cross_entropy_loss` are supported, :ref:`mse` is not supported.
-       - On GPU, ``resultsToCompute`` only computes ``value``, ``gradient``, and ``hessian``.
-   * - :ref:`logistic_regression`
-     - ``penaltyL1`` is not supported on GPU
+GPU support in DAAL part of library has been deprecated and removed from library.
```
Comment (aepanchi) [inline docs/source/daal/cpu-vs-gpu.rst:22]:
```suggestion
The GPU support in the DAAL part of library is deprecated and removed from the library.
```

File: docs/source/daal/cpu-vs-gpu.rst [hunk 5]
```diff
@@ -19,90 +19,5 @@
...
+Please use GPU support in oneDAL :ref:`_onedal_gpu_support`
```
Comment (aepanchi) [inline docs/source/daal/cpu-vs-gpu.rst:23]:
```suggestion
Use the :ref:`GPU support in oneDAL<onedal_gpu_support>`
```

File: docs/source/onedal/introduction.rst [hunk 6]
```diff
@@ -16,10 +16,51 @@
 
 Introduction
 ************
+oneDAL provides redesigned versions of interfaces that account for multi-device targets
```
Comment (aepanchi) [inline docs/source/onedal/introduction.rst:19]:
```suggestion
oneDAL provides redesigned versions of interfaces that account for the multi-device targets.
```

File: docs/source/onedal/introduction.rst [hunk 7]
```diff
@@ -16,10 +16,51 @@
...
+such as CPU and GPU, distributed SPMD interfaces and many more.
```
Comment (aepanchi) [inline docs/source/onedal/introduction.rst:20]:
```suggestion
For example, CPU and GPU, distributed SPMD interfaces, and many more.
```

File: docs/source/onedal/introduction.rst [hunk 8]
```diff
@@ -16,10 +16,51 @@
...
+
+Algorithms support
```
Comment (aepanchi) [inline docs/source/onedal/introduction.rst:22]:
```suggestion
Algorithms Support
```

File: docs/source/onedal/introduction.rst [hunk 9]
```diff
@@ -16,10 +16,51 @@
...
+   * - :ref:`_alg_dbscan`
+     - :ref:`batch <batch>`, :ref:`distributed <distributed>`
+   * - :ref:`_alg_basic_statistics`
+     - :ref:`batch <batch>`, :ref:`distributed <distributed>`
+
 
 .. toctree::
    :maxdepth: 1
 
+   gpu_support.rst
```
Comment (aepanchi) [inline docs/source/onedal/introduction.rst:63]: which file is that? could not find it

File: docs/source/onedal/introduction.rst [hunk 10]
```diff
@@ -16,10 +16,51 @@
 
 Introduction
 ************
+oneDAL provides redesigned versions of interfaces that account for multi-device targets
+such as CPU and GPU, distributed SPMD interfaces and many more.
+
+Algorithms support
+*****************
```
Comment (ethanglaser) [inline docs/source/onedal/introduction.rst:23]:
```suggestion
******************
```
From docbuild warning

File: docs/source/onedal/gpu_support.rst [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+.. *******************************************************************************/
+
+.. _onedal_gpu_support:
+
+CPU and GPU Support
+===================
+Not all computation modes, methods, and parameters are supported on both CPU and GPU.
+Differences in CPU and GPU support are listed below.
+
+GPU supported tragets
```
Comment (ethanglaser) [inline docs/source/onedal/gpu_support.rst:24]:
```suggestion
GPU supported targets
```

File: docs/source/onedal/gpu_support.rst [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+*********************
+oneDAL library targets Intel GPUs support only, while technically it should be feasible
+to run on other HW platforms asuming SYCL runtime availability for platform.
```
Comment (ethanglaser) [inline docs/source/onedal/gpu_support.rst:27]:
```suggestion
to run on other HW platforms assuming SYCL runtime availability for platform.
```

File: docs/source/daal/cpu-vs-gpu.rst [hunk 13]
```diff
@@ -19,90 +19,5 @@
...
-       - On GPU, the ``weights`` parameter is not supported.
-   * - :ref:`kernel`
-     - On GPU, the only supported computation mode (``ComputationMode``) is ``matrixMatrix``.
-   * - :ref:`objective_function`
-     -
-       - On GPU, only :ref:`logistic_loss` and :ref:`cross_entropy_loss` are supported, :ref:`mse` is not supported.
-       - On GPU, ``resultsToCompute`` only computes ``value``, ``gradient``, and ``hessian``.
-   * - :ref:`logistic_regression`
-     - ``penaltyL1`` is not supported on GPU
+The GPU support in the DAAL part of library is deprecated and removed from the library.
```
Comment (ethanglaser) [inline docs/source/daal/cpu-vs-gpu.rst:22]:
```suggestion
GPU support in the DAAL part of the library is deprecated and removed.
```

File: docs/source/onedal/gpu_support.rst [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+.. _onedal_gpu_support:
+
+CPU and GPU Support
+===================
+Not all computation modes, methods, and parameters are supported on both CPU and GPU.
+Differences in CPU and GPU support are listed below.
+
+GPU supported targets
+*********************
+oneDAL library targets Intel GPUs support only, while technically it should be feasible
```
Comment (ethanglaser) [inline docs/source/onedal/gpu_support.rst:26]:
```suggestion
oneDAL targets support for Intel GPUs only, while technically it should be feasible
```
Either "The oneDAL library targets..." or just "oneDAL targets..." would work, whichever is preferred.

File: docs/source/onedal/gpu_support.rst [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+     - 
+   * - :ref:`alg_pca`
+     - 
+   * - :ref: Logistic Regression
+     - 
+   * - :ref: Linear Regression
+     - 
+   * - :ref:`alg_df`
+     - 
+   * - :ref: Extra Trees
```
Comment (ethanglaser) [inline docs/source/onedal/gpu_support.rst:50]: Are these intended to be marked with `` (for Logistic, Linear, Extra Trees)?

File: docs/source/onedal/introduction.rst [hunk 16]
```diff
@@ -16,10 +16,51 @@
 
 Introduction
 ************
+oneDAL provides redesigned versions of interfaces that account for the multi-device targets.
```
Comment (ethanglaser) [inline docs/source/onedal/introduction.rst:19]:
```suggestion
oneDAL provides redesigned versions of interfaces that account for multi-device targets.
```

File: docs/source/onedal/gpu_support.rst (hunk 15 above)
Comment (napetrov) [reply docs/source/onedal/gpu_support.rst:50]: there should be documentation placed for those so we can link to them.

File: docs/source/onedal/gpu_support.rst [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+Differences in CPU and GPU support are listed below.
+
+GPU supported targets
+*********************
+oneDAL targets support for Intel GPUs only, while technically it should be feasible
+to run on other HW platforms assuming SYCL runtime availability for platform.
+
+.. tabularcolumns::  |\Y{0.5}|\Y{0.5}|
+
+.. list-table:: GPU support conciderations
```
Comment (aepanchi) [inline docs/source/onedal/gpu_support.rst:31]:
```suggestion
.. list-table:: GPU Support Considerations
```

File: docs/source/onedal/gpu_support.rst (hunk 15 above)
Comment (ethanglaser) [reply docs/source/onedal/gpu_support.rst:50]: What do these look like currently in the docs once they are built?

Comment (napetrov) [reply docs/source/onedal/gpu_support.rst:50]: image

But with ref word cleaned up - so no link

**Review by ethanglaser — APPROVED** [review]

**Review by aepanchi — APPROVED** [review]

**Pull request merged**